crates/pop-parachains/src/build/runtime/container_engine.rs

```diff
@@ -0,0 +1,106 @@
+// This file is copied and adapted from srtool-cli (https://github.com/chevdor/srtool-cli)
+// Original source: https://github.com/chevdor/srtool-cli/blob/master/cli/src/container_engine.rs
+
```

Comment:
As mentioned, this is a direct copy from the library with only a few modifications: removed println, added comments to the public enum (to avoid Clippy warnings), and included basic tests. I also opened an issue in the tool to migrate this into the library, allowing us to remove this file and use srtool-lib directly in the future.  https://github.com/chevdor/srtool-cli/issues/36